contracts/utils/MultiRateLimited.sol

```diff
@@ -0,0 +1,170 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "../refs/CoreRef.sol";
+import "./RateLimited.sol";
+import "./IMultiRateLimited.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+
+/// @title contract for putting a rate limit on how fast an address can perform an action e.g. Minting
+/// there are two buffers, one buffer which is each individual addresses's current buffer,
+/// and then there is a global buffer which is the buffer that each individual address must respect as well
+/// @author Fei Protocol
+contract MultiRateLimited is RateLimited, IMultiRateLimited {
+
+    /// @notice the rate per second for each address
+    mapping (address => uint256) public individualRateLimitPerSecond;
+
+    /// @notice the last time the buffer was used by each address 
+    mapping (address => uint256) public individualLastBufferUsedTime;
+
+    /// @notice the cap of the buffer that can be used at once by a single address
+    mapping (address => uint256) public individualBufferCap;
+
+    /// @notice the buffer at the timestamp of lastBufferUsedTime
+    mapping (address => uint256) private _bufferStored;
+
+    constructor(
+        address coreAddress,
+        uint256 _maxRateLimitPerSecond,
+        uint256 _rateLimitPerSecond,
+        uint256 _bufferCap,
+        uint256 maxBufferCap,
+        bool _doPartialAction
+    )
+        CoreRef(coreAddress)
+        RateLimited(_maxRateLimitPerSecond, _rateLimitPerSecond, _bufferCap, _doPartialAction)
+    {}
+
+    /// @notice add an authorized minter contract
+    /// @param _minter the new address to add as a minter
+    /// @param _rateLimitPerSecond the rate limit per second for this minter
+    /// @param _bufferCap  the buffer cap for this minter
+    function addAddress(address _minter, uint256 _rateLimitPerSecond, uint256 _bufferCap) external virtual override onlyGovernorOrAdmin {
```

Comment:
You're at a level of abstraction here where addresses are not necessarily "minters", but are rate-limited on anything. I wouldn't call the addresses "minter", because we could rate-limit redemptions or swaps or anything else

Reply:
what should we call them instead?

Reply:
`_throttledAddress`, `_rateLimitedAddress`?

Reply:
done